packages/dapp-svelte-wallet/api/src/types.js

```diff
@@ -8,7 +8,44 @@
  */
 
 /**
- * @typedef {string | string[]} Petname
+ * @typedef {string | string[]} Petname A petname can either be a plain string
+ * or a path for which the first element (the edgename) is a petname.  We are
+ * migrating away from using plain strings, for consistency.
+ */
+
+/**
+ * @typedef {Object} WalletBridge The wallet methods available within the
+ * context of a Dapp.
+ * @property {(offer: OfferState) => Promise<string>} addOffer
+ * @property {(offer: OfferState, invitation: ERef<Payment>) => Promise<string>}
+ * addOfferInvitation add an invitation to the specified offer
+ * @property {(brandBoardId: string) => Promise<string>} getDepositFacetId
+ * @property {() => Promise<Notifier<Array<PursesFullState>>>}
+ * getPursesNotifier
+ * @property {() => Promise<Notifier<Array<OfferState>>>}
+ * getOffersNotifier
+ * @property {(petname: Petname, issuerBoardId: string) => Promise<void>}
+ * suggestIssuer
+ * @property {(petname: Petname, installationBoardId: string) => Promise<void>}
+ * suggestInstallation
+ * @property {(petname: Petname, instanceBoardId: string) => Promise<void>}
+ * suggestInstance
+ */
+
+/**
+ * @typedef {Object} WalletUser the presence exposed as `local.wallet` (or
+ * `home.wallet`)
+ * @property {(suggestedDappPetname: Petname, dappOrigin: string) =>
+ * Promise<WalletBridge>} getBridge return a wallet bridge corresponding to an
+ * origin that must be approved in the wallet UI
+ * @property {() => Promise<WalletBridge>} getPreapprovedBridge return the
+ * wallet bridge that bypasses Dapp-authorization
```

Comment:
Is there a reason to give the user a preapprovedBridge rather than the admin?

Reply:
Could you add more contextual comments? It's helpful to know when, where and why something will be used. 

For example, when would a user call `getBridge` and when would a user call `getPreapprovedBridge`? Wouldn't the user instead probably want to call `getAdmin` and actually get the admin facet?

If the reason is so we don't need to type or document the admin facet, let's add a comment stating as such so we know there wasn't another design reason.

Reply:
I've documented these things now.